[WPE][GTK] Website data leaked outside specified data/cache directories

mcatanzaro opened the pull request:
#### 38ffac2857ee4b460882c7a67b1e3b63d949cb93
<pre>
[WPE][GTK] Website data leaked outside specified data/cache directories
<a href="https://bugs.webkit.org/show_bug.cgi?id=239521">https://bugs.webkit.org/show_bug.cgi?id=239521</a>

Reviewed by Adrian Perez de Castro.

Currently whenever somebody adds a new type of website data, you have to
add new public WPE/GTK API to WebKitWebsiteDataManager to allow
configuring the data location. If this step is not performed, the data
will leak outside the expected directory locations, which is a serious
privacy issue. Sadly, this step is usually missed in practice. We released
2.36 without accounting for the new general storage directory, resulting
in the general storage directory being created outside the base data
directory, which is not acceptable. Similarly, we missed
deviceidhashsalts, which was added *four years ago*. This case is a little
weird because we did add a WEBKIT_WEBSITE_DATA_DEVICE_ID_HASH_SALT
WebKitWebsiteDataTypes flag to allow deleting the data, but
unfortunately we missed that configuring a directory under the base data
directory was required.

This is bad. Apple developers shouldn&apos;t be expected to write Linux-specific
code when making cross-platform changes. Linux developers shouldn&apos;t need
to worry that oversights will lead to data being stored outside
permitted locations.

This commit attempts to fix this *comprehensively*.

First: we need to make sure the default directories only get used when
base data directory or base cache directory is NULL, and also ensure
this is hard to mess up. I considered many different approaches but
ultimately decided the simplest way to do it is to make all the default
directories relative to the base directories. This requires some
cross-platform changes in WebsiteDataStoreConfiguration, but nothing too
disruptive. I&apos;ve used a default null string parameter to minimize the
amount of code changes required in Apple ports, and left the base
directories unused for Apple. For Windows, I&apos;ve gone ahead and
implemented even though they are unused, because the implementation is
so simple. I&apos;ve also added an assert to make sure WPE/GTK always use the
base directories. By pushing the base directory logic from
WebKitWebsiteDataManager down to WebsiteDataStoreConfiguration, it&apos;s now
hopefully quite difficult to mess up, and the burden on Apple developers
is limited to just choosing whether the new directory should be data or
cache.

I&apos;ve also renamed WillCopyPathsFromExistingConfiguration to
ShouldInitializePaths, in order to better match the new semantics, and
brought it inside the class since it is no longer needed by the public
WebsiteDataStoreConfiguration::create methods.

Next problem: the default WebsiteDataStore. For cross-platform WebKit, I
suppose this can be used whenever a specific WebsiteDataStore is not
configured, and therefore it ought to be a persistent data store that
actually stores data on disk. This should never happen for WPE and GTK:
a specific WebsiteDataStore must always be used. This is currently
ensured by WebKitWebContext, which creates its own WebKitWebsiteDataStore
if the application does not do so itself. The default WebsiteDataStore
is still needed for prewarmed processes, but I hope for nothing else.
Accordingly, we can safely make it nonpersistent for WPE and GTK, but
not for other ports. This is required to comprehensively ensure there is
no way to create a WebsiteDataStoreConfiguration without initializing
the base data and base cache directories unless it is an ephemeral
store. In particular, we can assert the WebsiteDataStoreConfiguration
is either created with base directories, or ephemeral.

Finally, I got a bit carried away and decided to deprecate all the
non-base directory configuration APIs from WebKitWebsiteDataManager.
These are of very limited utility because frankly, who cares what each
particular subdirectory is named? What applications really want to do is
ensure that all data gets stored only where the application expects it
to be stored, a use case that can be achieved by configuring only the
base data and base cache directories. By deprecating these, we remove
the expectation that we should add more in the future or keep the list
of directories here updated, and we can remove them in the upcoming GTK
4 API version. I&apos;ve included some documentation updates to remove
references to the deprecated properties and functions, and snuck in a
few drive-by fixes. All of this could have been done separately, but it
makes sense to do here so as to eliminate the temptation to add new APIs
to configure the general storage directory and hash salts.

Future consideration: in future API versions, where
WebKitWebsiteDataManager&apos;s localstorage and IndexedDB directory
configuration APIs are removed, we can change
WebsiteDataStore::defaultShouldUseCustomStoragePaths to allow WPE and
GTK to use the new general storage directory instead of the legacy
custom localstorage and IndexedDB directories. Cannot do that yet
because the old APIs must still be supported even if deprecated.

* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreConfigurationRef.cpp:
(WKWebsiteDataStoreConfigurationCreate):
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
(webkitWebsiteDataManagerGetProperty):
(webkit_website_data_manager_class_init):
(webkitWebsiteDataManagerGetDataStore):
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.h.in:
* Source/WebKit/UIProcess/Inspector/gtk/WebInspectorUIProxyGtk.cpp:
(WebKit::inspectorWebsiteDataStore):
* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultAlternativeServicesDirectory):
(WebKit::WebsiteDataStore::defaultHSTSStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaCacheDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultDeviceIdHashSaltsStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::WebsiteDataStore::defaultJavaScriptConfigurationDirectory):
(WebKit::WebsiteDataStore::defaultModelElementCacheDirectory):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::defaultDataStoreIsPersistent):
(WebKit::WebsiteDataStore::defaultDataStore):
(WebKit::WebsiteDataStore::defaultMediaCacheDirectory):
(WebKit::WebsiteDataStore::defaultAlternativeServicesDirectory):
(WebKit::WebsiteDataStore::defaultJavaScriptConfigurationDirectory):
(WebKit::WebsiteDataStore::defaultDeviceIdHashSaltsStorageDirectory):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.cpp:
(WebKit::WebsiteDataStoreConfiguration::WebsiteDataStoreConfiguration):
(WebKit::WebsiteDataStoreConfiguration::initializePaths):
(WebKit::WebsiteDataStoreConfiguration::copy const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.h:
(WebKit::WebsiteDataStoreConfiguration::create):
(WebKit::WebsiteDataStoreConfiguration::createWithBaseDirectories):
* Source/WebKit/UIProcess/WebsiteData/playstation/WebsiteDataStorePlayStation.cpp:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Source/WebKit/UIProcess/WebsiteData/win/WebsiteDataStoreWin.cpp:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Source/WebKit/UIProcess/glib/WebsiteDataStoreGLib.cpp:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultHSTSStorageDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultDeviceIdHashSaltsStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::programName):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Tools/MiniBrowser/gtk/main.c:
(gotWebsiteDataCallback):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(testWebsiteDataConfiguration):
(testWebsiteDataEphemeral):
(testWebsiteDataITP):

Canonical link: <a href="https://commits.webkit.org/253824@main">https://commits.webkit.org/253824@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6209144d0f0d10c5ee5999731ce46a087f0b5236

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/87197 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/31284 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/18025 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/96226 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149770 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/91173 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29648 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/25878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/91209 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92813 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/68/builds/23951 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/74002 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/79313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/78927 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/79172 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/79313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/27373 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/13009 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/27320 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/14024 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2697 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/29005 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/36882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28944 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33301 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

aperezdc reviewed (approved):
r=me, you might want to wait a few hours before landing in case @carlosgcampos or @szewai may have anything to add here.

mcatanzaro committed:
[WPE][GTK] Website data leaked outside specified data/cache directories
https://bugs.webkit.org/show_bug.cgi?id=239521

Reviewed by Adrian Perez de Castro.

Currently whenever somebody adds a new type of website data, you have to
add new public WPE/GTK API to WebKitWebsiteDataManager to allow
configuring the data location. If this step is not performed, the data
will leak outside the expected directory locations, which is a serious
privacy issue. Sadly, this step is usually missed in practice. We released
2.36 without accounting for the new general storage directory, resulting
in the general storage directory being created outside the base data
directory, which is not acceptable. Similarly, we missed
deviceidhashsalts, which was added *four years ago*. This case is a little
weird because we did add a WEBKIT_WEBSITE_DATA_DEVICE_ID_HASH_SALT
WebKitWebsiteDataTypes flag to allow deleting the data, but
unfortunately we missed that configuring a directory under the base data
directory was required.

This is bad. Apple developers shouldn't be expected to write Linux-specific
code when making cross-platform changes. Linux developers shouldn't need
to worry that oversights will lead to data being stored outside
permitted locations.

This commit attempts to fix this *comprehensively*.

First: we need to make sure the default directories only get used when
base data directory or base cache directory is NULL, and also ensure
this is hard to mess up. I considered many different approaches but
ultimately decided the simplest way to do it is to make all the default
directories relative to the base directories. This requires some
cross-platform changes in WebsiteDataStoreConfiguration, but nothing too
disruptive. I've used a default null string parameter to minimize the
amount of code changes required in Apple ports, and left the base
directories unused for Apple. For Windows, I've gone ahead and
implemented even though they are unused, because the implementation is
so simple. I've also added an assert to make sure WPE/GTK always use the
base directories. By pushing the base directory logic from
WebKitWebsiteDataManager down to WebsiteDataStoreConfiguration, it's now
hopefully quite difficult to mess up, and the burden on Apple developers
is limited to just choosing whether the new directory should be data or
cache.

I've also renamed WillCopyPathsFromExistingConfiguration to
ShouldInitializePaths, in order to better match the new semantics, and
brought it inside the class since it is no longer needed by the public
WebsiteDataStoreConfiguration::create methods.

Next problem: the default WebsiteDataStore. For cross-platform WebKit, I
suppose this can be used whenever a specific WebsiteDataStore is not
configured, and therefore it ought to be a persistent data store that
actually stores data on disk. This should never happen for WPE and GTK:
a specific WebsiteDataStore must always be used. This is currently
ensured by WebKitWebContext, which creates its own WebKitWebsiteDataStore
if the application does not do so itself. The default WebsiteDataStore
is still needed for prewarmed processes, but I hope for nothing else.
Accordingly, we can safely make it nonpersistent for WPE and GTK, but
not for other ports. This is required to comprehensively ensure there is
no way to create a WebsiteDataStoreConfiguration without initializing
the base data and base cache directories unless it is an ephemeral
store. In particular, we can assert the WebsiteDataStoreConfiguration
is either created with base directories, or ephemeral.

Finally, I got a bit carried away and decided to deprecate all the
non-base directory configuration APIs from WebKitWebsiteDataManager.
These are of very limited utility because frankly, who cares what each
particular subdirectory is named? What applications really want to do is
ensure that all data gets stored only where the application expects it
to be stored, a use case that can be achieved by configuring only the
base data and base cache directories. By deprecating these, we remove
the expectation that we should add more in the future or keep the list
of directories here updated, and we can remove them in the upcoming GTK
4 API version. I've included some documentation updates to remove
references to the deprecated properties and functions, and snuck in a
few drive-by fixes. All of this could have been done separately, but it
makes sense to do here so as to eliminate the temptation to add new APIs
to configure the general storage directory and hash salts.

Future consideration: in future API versions, where
WebKitWebsiteDataManager's localstorage and IndexedDB directory
configuration APIs are removed, we can change
WebsiteDataStore::defaultShouldUseCustomStoragePaths to allow WPE and
GTK to use the new general storage directory instead of the legacy
custom localstorage and IndexedDB directories. Cannot do that yet
because the old APIs must still be supported even if deprecated.

* Source/WebKit/UIProcess/API/C/WKWebsiteDataStoreConfigurationRef.cpp:
(WKWebsiteDataStoreConfigurationCreate):
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.cpp:
(webkitWebsiteDataManagerGetProperty):
(webkit_website_data_manager_class_init):
(webkitWebsiteDataManagerGetDataStore):
* Source/WebKit/UIProcess/API/glib/WebKitWebsiteDataManager.h.in:
* Source/WebKit/UIProcess/Inspector/gtk/WebInspectorUIProxyGtk.cpp:
(WebKit::inspectorWebsiteDataStore):
* Source/WebKit/UIProcess/WebsiteData/Cocoa/WebsiteDataStoreCocoa.mm:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultAlternativeServicesDirectory):
(WebKit::WebsiteDataStore::defaultHSTSStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaCacheDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultDeviceIdHashSaltsStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::WebsiteDataStore::defaultJavaScriptConfigurationDirectory):
(WebKit::WebsiteDataStore::defaultModelElementCacheDirectory):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::defaultDataStoreIsPersistent):
(WebKit::WebsiteDataStore::defaultDataStore):
(WebKit::WebsiteDataStore::defaultMediaCacheDirectory):
(WebKit::WebsiteDataStore::defaultAlternativeServicesDirectory):
(WebKit::WebsiteDataStore::defaultJavaScriptConfigurationDirectory):
(WebKit::WebsiteDataStore::defaultDeviceIdHashSaltsStorageDirectory):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.h:
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.cpp:
(WebKit::WebsiteDataStoreConfiguration::WebsiteDataStoreConfiguration):
(WebKit::WebsiteDataStoreConfiguration::initializePaths):
(WebKit::WebsiteDataStoreConfiguration::copy const):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStoreConfiguration.h:
(WebKit::WebsiteDataStoreConfiguration::create):
(WebKit::WebsiteDataStoreConfiguration::createWithBaseDirectories):
* Source/WebKit/UIProcess/WebsiteData/playstation/WebsiteDataStorePlayStation.cpp:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Source/WebKit/UIProcess/WebsiteData/win/WebsiteDataStoreWin.cpp:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Source/WebKit/UIProcess/glib/WebsiteDataStoreGLib.cpp:
(WebKit::WebsiteDataStore::defaultApplicationCacheDirectory):
(WebKit::WebsiteDataStore::defaultNetworkCacheDirectory):
(WebKit::WebsiteDataStore::defaultCacheStorageDirectory):
(WebKit::WebsiteDataStore::defaultHSTSStorageDirectory):
(WebKit::WebsiteDataStore::defaultGeneralStorageDirectory):
(WebKit::WebsiteDataStore::defaultIndexedDBDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultServiceWorkerRegistrationDirectory):
(WebKit::WebsiteDataStore::defaultLocalStorageDirectory):
(WebKit::WebsiteDataStore::defaultMediaKeysStorageDirectory):
(WebKit::WebsiteDataStore::defaultDeviceIdHashSaltsStorageDirectory):
(WebKit::WebsiteDataStore::defaultWebSQLDatabaseDirectory):
(WebKit::WebsiteDataStore::defaultResourceLoadStatisticsDirectory):
(WebKit::programName):
(WebKit::WebsiteDataStore::cacheDirectoryFileSystemRepresentation):
(WebKit::WebsiteDataStore::websiteDataDirectoryFileSystemRepresentation):
* Tools/MiniBrowser/gtk/main.c:
(gotWebsiteDataCallback):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebsiteData.cpp:
(testWebsiteDataConfiguration):
(testWebsiteDataEphemeral):
(testWebsiteDataITP):

Canonical link: https://commits.webkit.org/253824@main
